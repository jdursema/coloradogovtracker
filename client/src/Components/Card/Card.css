.card {
  background: #fff;
  border-radius: 2px;
  height: 300px;
  width: 230px;
  box-shadow: 0 3px 6px rgba(0, 0, 0, 0.16), 0 3px 6px rgba(0, 0, 0, 0.23);
  margin: 10px;
  display: flex;
  flex-direction: column; }

  .card-head {
    display:flex;
    flex-direction: row;
    justify-content: space-between;
    margin: 5px;
  }

  .candidate-card-img {
    width: 60px;
    height: 60px;
    border-radius: 50%
  }

  .card-head-text {
    display:flex;
    flex-direction:column;
  }