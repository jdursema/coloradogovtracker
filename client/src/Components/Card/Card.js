import React, { Component } from 'react';
import { fetchIndividualContribution } from '../../Helper/helper';
import PropTypes from 'prop-types';

class Card extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showDetails: false,
      contributionDetails:{}
    };
  }
  
  toggleDetails = (async(recordId) => {
    /* eslint-disable */
    const cardDetails = await fetchIndividualContribution(recordId);
    /* eslint-enable */
    this.setState({contributionDetails: cardDetails});
    this.setState({showDetails: !this.state.showDetails});
  })


   capitalize = (string) =>{
     let lowercase = string.toLowerCase();
     return lowercase.charAt(0).toUpperCase() + lowercase.slice(1);
   };

   titleCase = (string) => {
     string = string.toLowerCase().split(' ');
     for (var i = 0; i < string.length; i++) {
       string[i] = string[i].charAt(0).toUpperCase() + string[i].slice(1); 
     }
     return string.join(' ');
   }

  formatDate = ( date) => {
    const newDate = new Date(date);
    const day = newDate.getDate();
    const month = newDate.getUTCMonth()+1;
    const year = newDate.getFullYear();

    return month + '/' + day + '/' + year;
  };

 label = (field, label) => {
   return field.length ? label : null;
 };

 formatAmount = (amount) => {
   let formattedNumber = (amount + "").replace(/\b(\d+)((\.\d+)*)\b/g, function(a, b, c) {
     return (b.charAt(0) > 0 && !(c || ".").lastIndexOf(".") ? b.replace(/(\d)(?=(\d{3})+$)/g, "$1,") : b) + c;
   });
   return formattedNumber;
 
 }

 render () {
   const {firstName, lastName, amount, recordId, type} = this.props;


   return (
     <div className = "contribution-card" onClick = {() => { this.toggleDetails(recordId); }}> 
       { 
         !this.state.showDetails &&
      <div className = "donation-side">
        <div className = "donor-row">
          <div className = "card-row-small-text">{this.label(lastName, 'DONOR')} </div>
          <div className = "card-row-medium-text">{firstName} {lastName}</div> 
        </div>
        <div className = "donation-money-row">
          <div className = "card-row-small-text">CONTRIBUTION</div>
          <div className = "card-row-big-number"> ${this.formatAmount(amount)} </div>
        </div>
        <div className = "donor-row">
          <div className = "card-row-small-text">CONTRIBUTION TYPE </div>
          <div className = "card-row-medium-text">{type}</div> 
        </div>
      </div>

       }
       {
         this.state.showDetails &&
      <div>
        <div className = "card-back-row">
          <div className = "card-row-small-text"> DONATION DATE </div>
          <div className = "card-row-medium-text">{this.formatDate(this.state.contributionDetails[0].contribution_date)} </div>
        </div>
        <div className = "card-back-row">
          <div className = "card-row-small-text"> DONOR ADDRESS </div>
          <div className = "card-row-medium-text">
            {this.state.contributionDetails[0].donor_city},
            {this.state.contributionDetails[0].donor_state}
            {this.state.contributionDetails[0].donor_zip}
          </div>
        </div>
        <div className = "card-back-row">
          <div className = "card-row-small-text"> {this.label(this.state.contributionDetails[0].donor_employer, 'DONOR EMPLOYER:')} </div>
          <div className = "card-row-medium-text">{this.titleCase(this.state.contributionDetails[0].donor_employer)}</div>
        </div>
        <div className = "card-back-row">
          <div className = "card-row-small-text"> {this.label(this.state.contributionDetails[0].donor_occupation, 'DONOR OCCUPATION:')} </div>
          <div className = "card-row-medium-text">{this.titleCase(this.state.contributionDetails[0].donor_occupation)}</div>
        </div>
      </div>
       }
     </div>
   );
 }
}

export default Card;

Card.propTypes = {
  id: PropTypes.string,
  firstName: PropTypes.string,
  lastName: PropTypes.string,
  amount: PropTypes.string,
  type: PropTypes.string,
  recordId: PropTypes.string


};