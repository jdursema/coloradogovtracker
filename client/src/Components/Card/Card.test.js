import Card from './Card';
import React from 'react';
import Enzyme, { shallow } from 'enzyme';



describe('card', () => {
  it('should match its snapshot', () => {


    const card = shallow(<Card
      id = {42}
      amount = {43}
      firstName = "Tommy"
      lastName = "Pickles"
      recordId = {6578998}
      date = "5/26/17 0:00"
      occupation = 'Director'
    />);

    expect(card).toMatchSnapshot();
  });


})

  