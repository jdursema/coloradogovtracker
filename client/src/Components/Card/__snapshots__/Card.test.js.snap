// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card should match its snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    amount={43}
    date="5/26/17 0:00"
    firstName="Tommy"
    id={42}
    lastName="Pickles"
    occupation="Director"
    recordId={6578998}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="donation-side"
>
          <div
                    className="donor-row"
          >
                    <div
                              className="card-row-small-text"
                    >
                              DONOR
                               
                    </div>
                    <div
                              className="card-row-medium-text"
                    >
                              Tommy
                               
                              Pickles
                    </div>
          </div>
          <div
                    className="donation-money-row"
          >
                    <div
                              className="card-row-small-text"
                    >
                              CONTRIBUTION
                    </div>
                    <div
                              className="card-row-big-number"
                    >
                               $
                              43
                               
                    </div>
          </div>
          <div
                    className="donor-row"
          >
                    <div
                              className="card-row-small-text"
                    >
                              CONTRIBUTION TYPE 
                    </div>
                    <div
                              className="card-row-medium-text"
                    />
          </div>
</div>,
        false,
      ],
      "className": "contribution-card",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="donor-row"
>
              <div
                            className="card-row-small-text"
              >
                            DONOR
                             
              </div>
              <div
                            className="card-row-medium-text"
              >
                            Tommy
                             
                            Pickles
              </div>
</div>,
            <div
              className="donation-money-row"
>
              <div
                            className="card-row-small-text"
              >
                            CONTRIBUTION
              </div>
              <div
                            className="card-row-big-number"
              >
                             $
                            43
                             
              </div>
</div>,
            <div
              className="donor-row"
>
              <div
                            className="card-row-small-text"
              >
                            CONTRIBUTION TYPE 
              </div>
              <div
                            className="card-row-medium-text"
              />
</div>,
          ],
          "className": "donation-side",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="card-row-small-text"
>
                  DONOR
                   
</div>,
                <div
                  className="card-row-medium-text"
>
                  Tommy
                   
                  Pickles
</div>,
              ],
              "className": "donor-row",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "DONOR",
                    " ",
                  ],
                  "className": "card-row-small-text",
                },
                "ref": null,
                "rendered": Array [
                  "DONOR",
                  " ",
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Tommy",
                    " ",
                    "Pickles",
                  ],
                  "className": "card-row-medium-text",
                },
                "ref": null,
                "rendered": Array [
                  "Tommy",
                  " ",
                  "Pickles",
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="card-row-small-text"
>
                  CONTRIBUTION
</div>,
                <div
                  className="card-row-big-number"
>
                   $
                  43
                   
</div>,
              ],
              "className": "donation-money-row",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "CONTRIBUTION",
                  "className": "card-row-small-text",
                },
                "ref": null,
                "rendered": "CONTRIBUTION",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " $",
                    "43",
                    " ",
                  ],
                  "className": "card-row-big-number",
                },
                "ref": null,
                "rendered": Array [
                  " $",
                  "43",
                  " ",
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="card-row-small-text"
>
                  CONTRIBUTION TYPE 
</div>,
                <div
                  className="card-row-medium-text"
/>,
              ],
              "className": "donor-row",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "CONTRIBUTION TYPE ",
                  "className": "card-row-small-text",
                },
                "ref": null,
                "rendered": "CONTRIBUTION TYPE ",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "card-row-medium-text",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="donation-side"
>
            <div
                        className="donor-row"
            >
                        <div
                                    className="card-row-small-text"
                        >
                                    DONOR
                                     
                        </div>
                        <div
                                    className="card-row-medium-text"
                        >
                                    Tommy
                                     
                                    Pickles
                        </div>
            </div>
            <div
                        className="donation-money-row"
            >
                        <div
                                    className="card-row-small-text"
                        >
                                    CONTRIBUTION
                        </div>
                        <div
                                    className="card-row-big-number"
                        >
                                     $
                                    43
                                     
                        </div>
            </div>
            <div
                        className="donor-row"
            >
                        <div
                                    className="card-row-small-text"
                        >
                                    CONTRIBUTION TYPE 
                        </div>
                        <div
                                    className="card-row-medium-text"
                        />
            </div>
</div>,
          false,
        ],
        "className": "contribution-card",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="donor-row"
>
                <div
                                className="card-row-small-text"
                >
                                DONOR
                                 
                </div>
                <div
                                className="card-row-medium-text"
                >
                                Tommy
                                 
                                Pickles
                </div>
</div>,
              <div
                className="donation-money-row"
>
                <div
                                className="card-row-small-text"
                >
                                CONTRIBUTION
                </div>
                <div
                                className="card-row-big-number"
                >
                                 $
                                43
                                 
                </div>
</div>,
              <div
                className="donor-row"
>
                <div
                                className="card-row-small-text"
                >
                                CONTRIBUTION TYPE 
                </div>
                <div
                                className="card-row-medium-text"
                />
</div>,
            ],
            "className": "donation-side",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="card-row-small-text"
>
                    DONOR
                     
</div>,
                  <div
                    className="card-row-medium-text"
>
                    Tommy
                     
                    Pickles
</div>,
                ],
                "className": "donor-row",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "DONOR",
                      " ",
                    ],
                    "className": "card-row-small-text",
                  },
                  "ref": null,
                  "rendered": Array [
                    "DONOR",
                    " ",
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Tommy",
                      " ",
                      "Pickles",
                    ],
                    "className": "card-row-medium-text",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Tommy",
                    " ",
                    "Pickles",
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="card-row-small-text"
>
                    CONTRIBUTION
</div>,
                  <div
                    className="card-row-big-number"
>
                     $
                    43
                     
</div>,
                ],
                "className": "donation-money-row",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "CONTRIBUTION",
                    "className": "card-row-small-text",
                  },
                  "ref": null,
                  "rendered": "CONTRIBUTION",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " $",
                      "43",
                      " ",
                    ],
                    "className": "card-row-big-number",
                  },
                  "ref": null,
                  "rendered": Array [
                    " $",
                    "43",
                    " ",
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="card-row-small-text"
>
                    CONTRIBUTION TYPE 
</div>,
                  <div
                    className="card-row-medium-text"
/>,
                ],
                "className": "donor-row",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "CONTRIBUTION TYPE ",
                    "className": "card-row-small-text",
                  },
                  "ref": null,
                  "rendered": "CONTRIBUTION TYPE ",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "card-row-medium-text",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
